syntax = "proto3";

package example;

service PromptService {
  rpc GetResponse (PromptRequest) returns (PromptReply) {};
  rpc GetResponseStream (PromptRequest) returns (stream PromptReply) {};
}

message PromptRequest {
  string prompt = 1;
}

message PromptReply {
  string answer = 1;
}

// Der Service, der die Funktion bereitstellt
service DatabaseHandlerService {
    // Die Funktion, die aufgerufen werden soll
    rpc GetSuggestions (GetSuggestionsRequest) returns (GetSuggestionsResponse);
    rpc GetAnswer (GetAnswerRequest) returns (GetAnswerResponse);
    rpc GetSuggestionsVector (GetSuggestionsRequest) returns (GetSuggestionsResponse);
    rpc GetAnswersQuestionsVector (GetSuggestionsRequest) returns (GetSuggestionsResponse);
    rpc GetSuggestionsVectorLocal (GetSuggestionsRequest) returns (GetSuggestionsResponse);
}

// Exakte Frage, zum beantworten
message GetAnswerRequest {
    string answer = 1;
    repeated string languages = 2;
    repeated string categories = 3;
}

// Antwort-Nachricht, die die Antwort enth채lt
message GetAnswerResponse {
    string answer = 1;
}

// Anfrage-Nachricht, die die erforderlichen Parameter enth채lt
message GetSuggestionsRequest {
    string topic = 1;
    repeated string languages = 2;
    repeated string categories = 3;
}

// Antwort-Nachricht, die die Vorschl채ge enth채lt
message GetSuggestionsResponse {
    repeated string suggestions = 1;
}